integrated_channels/moodle/client.py

```diff
@@ -311,38 +312,191 @@ def _wrapped_create_course_completion(self, user_id, payload):
+        """
+        serialized_data['wsfunction'] = 'core_course_create_courses'
+        announcement = serialized_data.pop('courses[0][announcement]')
+        course_format = serialized_data.pop('courses[0][format]')
+        response = self._wrapped_create_content_metadata(serialized_data)
+        # Response format should be [{"id":1, "shortname": "name"},{...}]
+        if response[0].get('id', None):
+            try:
+                post = self._create_forum_post(
+                    response[0].get('id'),
+                    announcement
+                )
+            except (TypeError, IndexError):
+                self._delete_content_metadata(response[0].get('id'))
+                raise ClientError(
+                    'Moodle Client Content Metadata Creation failed to create post for course {} '
+                    'as forum could not be found'.format(
+                        response[0].get('shortname'),
+                    ),
+                    HTTPStatus.NOT_FOUND.value
+                )
+            except Exception as exc:
+                self._delete_content_metadata(response[0].get('id'))
+                raise ClientError(
+                    'Moodle Client Content Metadata Creation failed to create post for course {} '
+                    'due to exception: {}'.format(response[0].get('shortname'), str(exc)),
+                    HTTPStatus.INTERNAL_SERVER_ERROR.value
+                )
+            if post.json().get('warnings', None) or post.json().get('exception', None):
+                self._delete_content_metadata(response[0].get('id'))
+                raise ClientError(
+                    'Moodle Client Content Metadata Creation failed to create post for course {}'
+                    .format(response[0].get('shortname')),
+                    HTTPStatus.BAD_REQUEST.value
+                )
+            format_params = {
+                'wsfunction': 'core_course_update_courses',
+                'courses[0][id]': response[0].get('id'),
+                'courses[0][format]': course_format,
+            }
+            course_update_response = self._post(format_params)
+            if course_update_response.json().get('warnings', None) or \
+                course_update_response.json().get('exception', None):
+                self._delete_content_metadata(response[0].get('id'))
+                raise ClientError(
+                    'Moodle Client Course Creation failed to update course format for course {}. '
+                    'Changes rolled back. '.format(response[0].get('shortname')),
+                    HTTPStatus.BAD_REQUEST.value
+                )
+        else:
+            raise ClientError(
+                'Moodle Client Course Creation failed to create course {}'
+                .format(serialized_data['courses[0][shortname]']),
+                HTTPStatus.BAD_REQUEST.value
+            )
+
+        return 200, ''
+
+
     def update_content_metadata(self, serialized_data):
```

Comment:
a doc string here explaining the flow may be nice, though you can decide if necessary.

Reply:
Yeah I'm still not done with doc strings (part of style fixes), but will make sure this section's is more detailed.